src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -369,19 +369,25 @@ def initialize_non_reference_person_immigrants(self, pop_data: SimulantData) ->
             ~existing_simulants["household_id"].isin(data_values.GQ_HOUSING_TYPE_MAP)
         ]["household_id"].unique()
 
-        # TODO: For now, we do a simple random sample of households.
-        # As part of future PUMA perturbation work, we plan to actually use
-        # the PUMA value in the ACS row in choosing the household they should join.
-        # NOTE: This does not do anything to state and PUMA, leaving them whatever they
-        # were for the ACS individual this immigrant is based on!
-        # This is obviously nonsensical, but mirrors the current behavior of domestic
-        # migration, where state and PUMA do not update on move.
-        new_simulants["household_id"] = vectorized_choice(
-            existing_household_ids,
-            n_to_choose=len(new_simulants),
-            randomness_stream=self.randomness,
-            additional_key="household_id",
-        ).astype(existing_simulants["household_id"].dtype)
+        if len(existing_household_ids) == 0:
+            # Extremely rare edge case when there are no existing households to assign to.
+            # Should only ever happen with very small population sizes.
+            # We just create new households and make the immigrants the reference people.
+            household_ids = self.households.create_households(
+                num_households=len(new_simulants),
+            )
+            new_simulants["household_id"] = household_ids
+            new_simulants["relation_to_household_head"] = "Reference person"
+        else:
+            # TODO: For now, we do a simple random sample of households.
+            # As part of future PUMA perturbation work, we plan to actually use
+            # the PUMA value in the ACS row in choosing the household they should join.
+            new_simulants["household_id"] = vectorized_choice(
+                existing_household_ids,
+                n_to_choose=len(new_simulants),
+                randomness_stream=self.randomness,
+                additional_key="household_id",
+            )
```

Comment:
`astype` is no longer needed?

Reply:
We do `.astype(int)` on the household_id column just below. This `astype` was actually doing nothing, because the type of `existing_simulants["household_id"]` was float anyway.